src/types/Graph.ts

```diff
@@ -82,3 +82,94 @@ export interface CytoscapeBaseEvent {
 export interface CytoscapeClickEvent extends CytoscapeBaseEvent {}
 export interface CytoscapeMouseInEvent extends CytoscapeBaseEvent {}
 export interface CytoscapeMouseOutEvent extends CytoscapeBaseEvent {}
+
+// Graph Structures
+export interface ProtocolTraffic {
+  protocol: string;
+  rates: any;
+}
+
+export interface GraphNodeData {
+  id: string;
+  parent?: string;
+  nodeType: NodeType;
+  namespace: string;
+  workload?: string;
+  app?: string;
+  version?: string;
+  service?: string;
+  destServices?: any;
+  traffic?: ProtocolTraffic[];
+  hasCB?: boolean;
+  hasMissingSC?: boolean;
+  hasVS?: boolean;
+  isDead?: boolean;
+  isGroup?: string;
+  isInaccessible?: boolean;
+  isMisconfigured?: string;
+  isOutside?: boolean;
+  isRoot?: boolean;
+  isServiceEntry?: string;
+  isUnused?: boolean;
+}
+
+export interface GraphEdgeData {
+  id: string;
+  source: string;
+  target: string;
+  traffic?: ProtocolTraffic;
```

Comment:
Same comment as above but also not that each edge can have at most one protocol field and it is expected to be set. It seems you removed it but if you look at the UI code you will see it is used in several places.